### ISSUE TITLE
0400s: SB: back by popular demand: and tweaks

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [ghacksuserjs/ghacks-user.js#803]
well nah, not by demand (but be honest, some of you bitched) 😀 

Since users are (supposed to be) adults, and the prefs are inactive, and there's a warning/note about it: i.e the risk in on the end-user.... and because information is power, I am willing to put back in the prefs for the SB UI

I'd like to make it a single numbered item: but for clarity, I'll leave it split here for now: ignore the numbers.

note: 0410 seems to be a master to 0411 which seems to be a master for 0412 (just look at the UI options getting greyed out as you change them): but I'm not interested in that aspect

tested in FF68: these are the prefs that change
note: so it's easier to read I have removed the `//` lines **just for this post**
```js
/* 0410: disable "Block dangerous and deceptive content"
 * This covers deceptive sites such as phishing and social engineering
 * [SETTING] Privacy & Security>Security>Deceptive Content and Software Protection ***/
    user_pref("browser.safebrowsing.malware.enabled", false);
    user_pref("browser.safebrowsing.phishing.enabled", false); // [FF50+]

/* 0411: disable "Block dangerous downloads"
 * This covers malware and PUPs (potentially unwanted programs)
 * [SETTING] Privacy & Security>Security>Deceptive Content and Software Protection ***/
    user_pref("browser.safebrowsing.downloads.enabled", false);

/* 0412: disable "Warn you about unwanted and uncommon software"
 * [SETTING] Privacy & Security>Security>Deceptive Content and Software Protection ***/
    user_pref("browser.safebrowsing.downloads.remote.block_potentially_unwanted", false);
    user_pref("browser.safebrowsing.downloads.remote.block_uncommon", false);
```

~~In `0412` above: those two prefs were not in our original (I took a v65 user.js): AFAIK we have never included those~~. The prefs we used to list were
```js
   // user_pref("browser.safebrowsing.downloads.remote.block_potentially_unwanted", false); // [FF48+]
   // user_pref("browser.safebrowsing.downloads.remote.block_uncommon", false); // [FF48+]
   // user_pref("browser.safebrowsing.downloads.remote.block_dangerous", false); // [FF49+]
   // user_pref("browser.safebrowsing.downloads.remote.block_dangerous_host", false); // [FF49+]
```

Tor Browser (based on ESR68) does not change these four prefs we used to use.instead relying on the "master" pref aspect (which I don't care about).

So I think the above info is good to go in some sort of PR.

-----

the current `0402`: Tor Browser also disable the URL. I'd like to add that

```js
/* 0402: disable binaries NOT in Safe Browsing local lists being checked
 * This is a real-time check with Google services
 * [SETUP-SECURITY] If you do not understand this, or if you want this protection, then override it ***/
user_pref("browser.safebrowsing.downloads.remote.enabled", false);
user_pref("browser.safebrowsing.downloads.remote.url", ""); // <-- ADDED
```

---

Honk if you agree (or type something). Also @earthlng

**Comment by rusty-snake** [comment]
> note: 0410 seems to be a master to 0411 which seems to be a master for 0412 (just look at the UI options getting greyed out as you change them): but I'm not interested in that aspect

Your right, I investigated this for my overrides,

```javascript
user_pref("browser.safebrowsing.malware.enabled", false);
user_pref("browser.safebrowsing.phishing.enabled", false); 
```

disables all connections to SB-Servers from google.

**Comment by Thorin-Oakenpants** [comment]
Thanks. I wonder why TB also flip `0411`: but it doesn't matter. I'll list them individually just in case someone wants to flip the bottom one only `¯\_(ツ)_/¯ `

**Comment by Thorin-Oakenpants** [comment]
Damnit: threw away all my digital notes. I'll have to retest (or I misread my results). I've listed the same prefs in the new draft as were in the old one: but then said they were different

**Comment by earthlng** [comment]
AFAICT 0410 are the master prefs for SB. (with the `.malware.` one being the master pref for the whole `.downloads.*` part)

After that, 0411 seems to be the master pref for the whole `browser.safebrowsing.downloads.*` branch.
Within that, `browser.safebrowsing.downloads.remote.enabled` (= the remote lookup part) is the master switch for the 4 `*.downloads.remote.block_*` prefs, 2 of which (`.block_dangerous` and `.block_dangerous_host`) are not available in the UI.

So the downloads/binary-checks structure looks like this (AFAICT!):

* `browser.safebrowsing.downloads.enabled`
  * `browser.safebrowsing.downloads.remote.enabled`
    * `browser.safebrowsing.downloads.remote.block_potentially_unwanted`
    * `browser.safebrowsing.downloads.remote.block_uncommon`
    * `browser.safebrowsing.downloads.remote.block_dangerous`
    * `browser.safebrowsing.downloads.remote.block_dangerous_host`

**Comment by Thorin-Oakenpants** [comment]
there's also `browser.safebrowsing.passwords.enabled` (default false), so it'll be interesting to see how that ties in if/when its flipped

**Comment by Thorin-Oakenpants** [comment]
I edited my first post re the "new" prefs. I will blame being tired and overworked, a lack of coffee, the new about:config layout in Nightly (it's ugly AF, spaced out, less legible esp with long prefs that wrap lines, and less functional) during my testing, and having "match whole word only" checked temporarily in my IDE

All that remains is how to format this. I prefer a single numbered item. No reference to the individual checkboxes, adding all the prefs (inactive)

```js
/* xxxx: disable Safe Browsing
 * [WARNING] Do this at your own risk
 * [SETTING] Privacy & Security>Security>Deceptive Content and Software Protection ***/
    user_pref("browser.safebrowsing.malware.enabled", false);
    user_pref("browser.safebrowsing.phishing.enabled", false);
    user_pref("browser.safebrowsing.downloads.enabled", false);
    user_pref("browser.safebrowsing.downloads.remote.block_potentially_unwanted", false);
    user_pref("browser.safebrowsing.downloads.remote.block_uncommon", false);
    user_pref("browser.safebrowsing.downloads.remote.block_dangerous", false);
    user_pref("browser.safebrowsing.downloads.remote.block_dangerous_host", false);
```

and add the extra pref to 0402. Class, discuss!

**Comment by earthlng** [comment]
```js
/** SAFE BROWSING (SB)
    This sub-section has been redesigned to differentiate between "real-time"/"user initiated"
    data being sent to Google from all other settings such as using local blocklists/whitelists and
    updating those lists. There are NO privacy issues here. *IF* required, a full url is never sent
    to Google, only a PART-hash of the prefix, and this is hidden with noise of other real PART-hashes.
    Google also swear it is anonymized and only used to flag malicious sites/activity. Firefox
    also takes measures such as striping out identifying parameters and storing safe browsing
    cookies in a separate jar. (#Turn on browser.safebrowsing.debug to monitor this activity)
    #Required reading [#] https://feeding.cloud.geek.nz/posts/how-safe-browsing-works-in-firefox/
    [SETTING] Privacy & Security>Security>Deceptive Content and Software Protection
    [1] https://wiki.mozilla.org/Security/Safe_Browsing
***/
/* 0410: disable Safe Browsing ("Block dangerous and deceptive content" in the UI)
 * [WARNING] Do this at your own risk! ***/
   // user_pref("browser.safebrowsing.malware.enabled", false);
   // user_pref("browser.safebrowsing.phishing.enabled", false);
/* 0411: disable checks for binary files/downloads (both local lookups + remote) ("Block dangerous downloads" in the UI) ***/
   // user_pref("browser.safebrowsing.downloads.enabled", false);
/* 0412: disable remote lookup checks for binary files/downloads (Google real-time checking) ***/
user_pref("browser.safebrowsing.downloads.remote.enabled", false);
user_pref("browser.safebrowsing.downloads.remote.url", "");
/* 0412b: don't "Warn me about unwanted and uncommon software" (part of remote lookup checks) ***/
   // user_pref("browser.safebrowsing.downloads.remote.block_potentially_unwanted", false);
   // user_pref("browser.safebrowsing.downloads.remote.block_uncommon", false);
/* 0419: disable 'ignore this warning' on Safe Browsing warnings
 * If clicked, it bypasses the block for that session. This is a means for admins to enforce SB
 * [TEST] see github wiki APPENDIX A: Test Sites: Section 5
 * [1] https://bugzilla.mozilla.org/1226490 ***/
   // user_pref("browser.safebrowsing.allowOverride", false);
```

IMO it should be something like this because this covers everyone's needs:

- people who don't want SB -> activate 0410
- people who want SB only for sites but don't want downloads checked -> activate 0411
- people want SB but no real-time checks with Evil Corp for binaries -> our default
- people want SB + remote checks but only for "dangerous" files and files from "dangerous hosts" and not be warned about unwanted and uncommon software -> comment out 0412 + activate 0412b

IMO we don't need to list the 2 `.remote.block_dangerous*` prefs because I don't see why anyone would only want to be warned about "unwanted and uncommon software" and not actual dangerous files and files from dangerous hosts; that just doesn't make any sense at all.

ps: allowOverride=04**19** as the last number in 041x to leave space for future additions (like, you mentioned passwords fe)

**Comment by Thorin-Oakenpants** [comment]
why do we even need 0412b if the master switch is already active in 0412? What a mess

**Comment by Thorin-Oakenpants** [comment]
- I've changed the wiki implementation page to refer to `0412`
- scratchpad
   - I can remove the items we put back
   - or I can leave them in and add `// re-added in 70-beta`
   - or I can do nothing
   - what do? @earthlng - I think it's fine to just rip them out of the scratchpad
- I'm going to do a ~~PR~~ commit in a minute or two
- If you're happy let me know and clap your hands ... if not you know where to find me

Notes: I based this on your structure in the previous comment. I want the individual settings in there. And I added in the pref structure, kinda. I'm sure you'll find something that needs a tweak
